fail2ban/server/filter.py

```diff
@@ -518,19 +526,45 @@ def findFailure(self, tupleLine, date=None, returnRawHost=False,
 						 % ("\n".join(failRegex.getMatchedLines()), timeText))
 				else:
 					self.__lineBuffer = failRegex.getUnmatchedTupleLines()
+					# retrieve failure-id, host, etc from failure match:
+					raw = returnRawHost
 					try:
-						host = failRegex.getHost()
-						if returnRawHost:
-							failList.append([failRegexIndex, IPAddr(host), date,
-								 failRegex.getMatchedLines()])
+						fail = failRegex.getGroups()
+						# failure-id:
+						fid = fail.get('fid')
+						# ip-address or host:
+						host = fail.get('ip4')
```

Comment:
But pro rata you are right here, following code looks better in this case, so I will add a commit for it:
```diff
-			host = fail.get('ip4')
+			host = fail.get('ip4') or fail.get('ip6')
 			if host is not None:
 				raw = True
 			else:
-				host = fail.get('ip6')
-				if host is not None:
-					raw = True
-				else:
```